tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFile.java

```diff
@@ -1497,22 +1544,38 @@ public Iterator<String> iterator(String prefix, Comparator<String> comparator) {
         return orderedSet.iterator();
     }
 
+    public Iterable<String> iterableDefault() {
+        if (DEFAULT_ITERATION_INCLUDES_EXTRAS) {
+            // same as fullIterable except skip paths with null values
+            Iterable<String> it = new FullIterable(this);
+            return () -> Iterators.filter(it.iterator(), p -> getStringValue(p) != null);
+        } else {
+            return iterableWithoutExtras();
+        }
+    }
+
+    private Iterable<String> iterableWithoutExtras() {
+        return this::iteratorWithoutExtras;
+    }
+
     public Iterable<String> fullIterable() {
         return new FullIterable(this);
     }
 
-    public static class FullIterable implements Iterable<String>, SimpleIterator<String> {
+    private static class FullIterable implements Iterable<String>, SimpleIterator<String> {
         private final CLDRFile file;
-        private final Iterator<String> fileIterator;
+        private final Iterator<String> fileIterator; // dataSource only
         private Iterator<String> extraPaths;
 
         FullIterable(CLDRFile file) {
             this.file = file;
-            this.fileIterator = file.iterator();
+            this.fileIterator = file.dataSource.iterator(); // file.iteratorWithoutExtras();
```

Comment:
```suggestion
            this.fileIterator = file.iteratorWithoutExtras();
```

would it be better to use the function for documentation?

Reply:
Good idea. And I'll rename `fileIterator` to `iteratorWithoutExtras` to make it clear and consistent.

Reply:
I've done that, in the 2nd commit